gradle/validation/jar-checks.gradle

```diff
@@ -122,17 +122,24 @@ subprojects {
       }
 
       def visited = new HashSet<>()
+      def seenDeps = new HashSet<>()
       def infos = []
 
       while (!queue.isEmpty()) {
         def dep = queue.removeFirst()
+        seenDeps.add(dep)
 
         // Skip any artifacts from other Solr Modules (they will be resolved there).
         if (dep.moduleGroup == "org.apache.solr") {
           continue
         }
 
-        queue.addAll(dep.children)
+        // Make sure we don't keep visiting the same children over and over again
```

Comment:
For completeness here is some debugging done in ASF slack but the root cause is grpc having a cyclic dependency when we don't account for runtime and compile dependencies separately.

https://mvnrepository.com/artifact/io.grpc/grpc-core/1.58.0
depends on grpc-util as runtime dependency
https://mvnrepository.com/artifact/io.grpc/grpc-util/1.58.0
depends on grpc-core as compile dependency
and grpc-util is new in 1.58.0 - https://mvnrepository.com/artifact/io.grpc/grpc-util

https://github.com/grpc/grpc-java/issues/10576

the issue that jar checks everything - both runtime and compile so thats the loop issue.
based on the comments its not REALLY an issue since compile and runtime can coexist like that but its just our jar-checks that has an issue w/ it

Reply:
This same thing exists in https://github.com/apache/lucene/blob/main/gradle/validation/jar-checks.gradle#L115 so will open a PR there

Reply:
https://github.com/apache/lucene/pull/12618